hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/AbstractMetadataStore.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl.recordstore;
+
+import com.hazelcast.internal.serialization.Data;
+import com.hazelcast.query.impl.AbstractMetadata;
+
+/**
+ * Abstraction of Metadata Store.
+ */
+public interface AbstractMetadataStore {
```

Comment:
```suggestion
public interface JsonMetadataStore {
```

Reply:
Addressed in 1ca96b2 and hazelcast/hazelcast@b573cf6